### ISSUE TITLE
Entire `switch` case missed if testing string literal

### ISSUE THREAD

**Issue by nator** [edvardchen/eslint-plugin-i18next#47]
I think this is caused by the fix for #2 , where switch cases statements with a literal test cause the entire block to be skipped. I have code like this that I'd like to fail the linter, but I don't quite see how to fix it and still keep #2 intact.

e.g. this title assignment shouldn't be allowed:
```ts
    switch (task.state) {
        case kTaskState_InProgress:
            style = classes.inProgress;
            title = "In Progress"
            break;
```
Thanks!

**Comment by nator** [comment]
I'm wrong, it's not skipping the body of the case statement. It seems like this test in no-literal-string.js might be ignoring "normal" assignment:
```js
        // var a: 'abc' = 'abc'
        if (typeObj.isStringLiteral()) {
          return;
        }
```
With that statement this second assignment is allowed, IMO it shouldn't be:
```ts
let title: string = i18next.t('common:test')
title = "omg lol"
```
I'm afraid I don't understand typescript well enough yet to know which is wrong or what the right fix is :(

**Comment by edvardchen** [comment]
@nator it's weird. I will check on it
